File: CH2-modules/EX5-body-mass-index/source-code/bodyMassIndex.js
const input = require("prompt-sync")();

function getWeightHeight(weight, height) {
  weight.value = input("Enter a weight (lb)     :");
  height.value = input("Enter a height (Inches) :");
}

function processBMI(weight, height, BMI) {
  BMI.value = (weight.value * 703) / (height.value * height.value);
}

function printBMI(BMI) {
  console.log("BMI :" + BMI.value.toFixed(1));
}

function main() {
  let weight = {
    value: 0,
  };
  let height = {
    value: 0,
  };
  let BMI = {
    value: 0,
  };

  getWeightHeight(weight, height);
  processBMI(weight, height, BMI);
  printBMI(BMI);
}

main();

File: CH2-modules/EX5-body-mass-index/source-code/bodyMassIndex.cpp
#include <iostream>
#include <iomanip>
using namespace std;

void getWeightHeight(float &weight, float &height)
{

  cout << "Enter a weight (lb) : ";
  cin >> weight;

  cout << "Enter a height (inches) : ";
  cin >> height;
}

void processBMI(float &weight, float &height, float &BMI)
{
  BMI = (weight * 703) / (height * height);
}

void printBMI(float &BMI)
{
  cout << fixed << setprecision(1);

  cout << "BMI : " << BMI << endl;
}

int main()
{
  float weight;
  float height;
  float BMI;

  getWeightHeight(weight, height);
  processBMI(weight, height, BMI);
  printBMI(BMI);
}